clippy_lints/src/slow_vector_initialization.rs

```diff
@@ -134,6 +155,14 @@ impl SlowVectorInit {
         }
     }
 
+    /// Checks if the given expression is `Vec::new()`
+    fn is_vec_new(cx: &LateContext<'_>, expr: &Expr<'_>) -> bool {
```

Comment:
nit: This is only used in `as_vec_initializer` afaict, if it's refactored into a two-variant enum then this should probably be inlined as naming it wouldn't clarify much then

Reply:
True, though I think we should then also do the same for the other already-existing method that checks for `Vec::with_capacity`, as it would be inconsistent/weird if only `is_vec_new` was inlined imo

Reply:
err, now actually getting to implement it, that `is_vec_with_capacity` function has slightly more stuff going on so it might be better to leave it

Reply:
Hmm.. It doesn't seem like what it does is necessary, that could probably be changed to matching on the path directly (rather than the method name plus the type).